test/exqlite/connection_test.exs

```diff
@@ -184,6 +184,19 @@ defmodule Exqlite.ConnectionTest do
 
       assert :ok == Connection.disconnect(nil, conn)
     end
+
+    test "executes before_disconnect before disconnecting" do
+      {:ok, conn} =
+        Connection.connect(
+          database: :memory,
+          before_disconnect: fn err, db ->
+            assert err
+            assert is_function(db.before_disconnect)
+          end
+        )
+
+      assert :ok == Connection.disconnect(true, conn)
```

Comment:
I _think_ what we want to do here as well is spin up a temporary `Agent` initialize it to 0, then inside the before disconnect, bump it +1, then assert that the value in the agent is 1 after disconnect is called.

I don't know if the concept of test spys are available in ex unit.